adapter/internal/api/apis_impl.go

```diff
@@ -151,6 +174,31 @@ func extractAPIProject(payload []byte) (apiProject ProjectAPI, err error) {
 	return apiProject, nil
 }
 
+func verifyMandatoryFields(apiJSON config.APIJsonData) error {
+	var errMsg string = ""
+	if apiJSON.Data.APIName == "" {
+		errMsg = "API Name cannot be empty"
+	}
+
+	if apiJSON.Data.APIVersion == "" {
+		errMsg = "API Version cannot be empty"
+	}
+
+	if apiJSON.Data.APIContext == "" {
+		errMsg = "API Context cannot be empty"
+	}
+
+	if apiJSON.Data.EndpointConfig.ProductionEndpoints.Endpoint == "" &&
+		apiJSON.Data.EndpointConfig.SandBoxEndpoints.Endpoint == "" {
+		errMsg = "API production and sandbox enpoints both cannot be empty"
+	}
+
+	if errMsg != "" {
+		return errors.New(errMsg)
+	}
```

Comment:
What happens if more than one fiellds are empty. Then only the lastly assigned error message will be logged.

Reply:
Its not logged, its throw the error and stop proceed. Do we need to show everything?

Reply:
append multiple fileds to message https://github.com/wso2/product-microgateway/pull/1785/commits/3705a414ed360024c9d0351580caed292bc7035e